client/state/sites/hooks/use-site-option.ts

```diff
@@ -0,0 +1,20 @@
+import { useSelector } from 'react-redux';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import getSiteOption from '../selectors/get-site-option';
+
+/**
+ * Returns a site option for a site
+ *
+ * @param {string} optionName The option you want to query
+ * @returns {*} The value of that option or null
```

Comment:
The jsdoc style type annotations are necessary in TypeScript, because the TypeScript types override them anyway.

```suggestion
 * @param optionName The option you want to query
 * @returns The value of that option or null
```